contracts/Wooracle.sol

```diff
@@ -37,16 +37,24 @@ pragma experimental ABIEncoderV2;
 
 import './libraries/InitializableOwnable.sol';
 import './interfaces/IWooracle.sol';
+import '@openzeppelin/contracts/math/SafeMath.sol';
 
 /// @title Wooracle implementation
 /// @notice Will be maintained and updated periodically by Woo.network in multichains.
 contract Wooracle is InitializableOwnable, IWooracle {
+
+    using SafeMath for uint256;
+
     /* ----- State variables ----- */
 
-    mapping(address => uint256) public prices;
-    mapping(address => uint256) public coeffs;
-    mapping(address => uint256) public spreads;
-    mapping(address => bool) public isValid;
+    // NOTE: decimal = 8 for all the fields
+    struct TokenInfo {
+        uint64 price;
+        uint64 coeff;
+        uint64 spread;
+    }
```

Comment:
isValid be set and only be set to false when the price is 0; so in logic,  isValid === (price == 0). So isValid field is no longer needed.